conans/test/functional/toolchains/cmake/test_cmake.py

```diff
@@ -36,8 +36,11 @@ def generate(self):
                 tc.variables.debug["MYVAR2_CONFIG"] = "MYVAR2_DEBUG"
                 tc.variables.release["MYVAR2_CONFIG"] = "MYVAR2_RELEASE"
                 tc.preprocessor_definitions["MYDEFINE"] = "MYDEF_VALUE"
+                tc.preprocessor_definitions["MYDEFINEINT"] = 42
                 tc.preprocessor_definitions.debug["MYDEFINE_CONFIG"] = "MYDEF_DEBUG"
                 tc.preprocessor_definitions.release["MYDEFINE_CONFIG"] = "MYDEF_RELEASE"
+                tc.preprocessor_definitions.debug["MYDEFINEINT_CONFIG"] = 421
```

Comment:
do we need this? I though you planned to remove that

Reply:
I have not yet fully decided, on one hand it is clear there is demand, on the other, it could be a pure build-system thing, and Conan agnostic about it. In the meantime, it totally makes sense that users want to provide integer preprocessor definitions, as they are very common. Maybe @guiserle could elaborate a bit further about their use case, and why such preprocessor definitions are not integrated in the build system scripts directly.

Reply:
First of all, thanks a lot for considering my request.

My use case is to define configuration values for libraries such as FreeRTOS or MCU SDK packages. Currently we define conan options for such values and pass those values to cmake on conan `build()` method. However that means for every configuration value (and these libraries have lots of them) we need to add logic on both conanfile.py and cmake files.

On top of that we are using Clion as IDE and we need to add these cmake variables (and others such as version numbers) on the IDE project settings manually. I was thinking about setting these values directly on `generate()` method so they would all be included in toolchain file and it would only be necessary to add the toolchain as value for CMAKE_TOOLCHAIN_FILE.

If that is not the intended use case for the generate method and this should be implemented on another way, I will understand it. Also I have considered the implementation of a custom generator for my particular use case if necessary

Reply:
I am going to merge this, seems a valid use case, thanks for the feedback.

Just beware that ``self.options`` are also always strings, even if you defined them as ``option: [1, 2, 3]``, internally, they are always strings. So you might need to convert it to ints, if you use it from there like ``cmake.preprocessor_defintions["MYDEF"] = int(self.options.myoption)``

Reply:
That's great. Thanks for taking my use case into account!